projects/plugins/jetpack/extensions/blocks/ai-paragraph/edit.js

```diff
@@ -69,13 +68,62 @@ function getSuggestionFromOpenAI(
 		} );
 }
 
+/**
+ * Gather data available in Gutenberg and prepare the best prompt we can come up with.
+ *
+ * @param {Function} select - function returning Gutenberg data store.
+ * @returns {string} - prompt ready to pipe to OpenAI.
+ */
+function preparePromptBasedOnEditorState( select ) {
+	const prompt = formatEditorCanvasContentForCompletion( select( 'core/block-editor' ) );
+
+	// If a user started typing something, we will just create a completion.
+	if ( prompt.length > 0 ) {
+		// We only take the last 240 chars into account, otherwise the prompt gets too long and because we have a 110 tokens limit, there is no place for response.
+		return prompt.slice( -240 );
+	}
+
+	// Let's grab post data so that we can do something smart.
+	const currentPost = select( 'core/editor' ).getCurrentPost();
+	if ( ! currentPost ) {
+		return '';
+	}
+
+	// If there is no title, there is not much we can do.
+	if ( ! currentPost.title ) {
+		return '';
+	}
+
+	// User did not set any categories, we are going to base the suggestions off a title.
+	if ( ! currentPost.categories ) {
+		/** translators: This will be a prompt to OpenAI to generate a post based on the post title */
+		return sprintf( __( 'Write me a post titled "%s"', 'jetpack' ), currentPost.title );
```

Comment:
I think the issue with that prompt and the one below is that OpenAI will include the title in the beginning of its response, so you end up with a double title. I'm not sure how to work around that, maybe the prompt should ask not to include the title of the post in the response? Would that work?

<img width="772" alt="image" src="https://user-images.githubusercontent.com/426388/212310335-af273a73-3347-4e78-b8d6-fef51c064981.png">

Reply:
I solved it bu asking specifically about content

This is where real money is in this - tailoring prompts to specific demands :)

Nice discovery!